src/Storage/FlysystemStorage.php

```diff
@@ -5,25 +5,31 @@
 use League\Flysystem\FileNotFoundException;
 use League\Flysystem\FilesystemInterface;
 use League\Flysystem\MountManager;
+use Psr\Container\ContainerInterface;
 use Symfony\Component\HttpFoundation\File\UploadedFile;
 use Vich\UploaderBundle\Mapping\PropertyMapping;
 use Vich\UploaderBundle\Mapping\PropertyMappingFactory;
 
 /**
  * @author Markus Bachmann <markus.bachmann@bachi.biz>
+ * @author Titouan Galopin <galopintitouan@gmail.com>
  */
 class FlysystemStorage extends AbstractStorage
 {
     /**
-     * @var MountManager Flysystem mount manager
+     * @var MountManager|ContainerInterface a registry to get FilesystemInterface instances
      */
-    protected $mountManager;
+    protected $registry;
 
-    public function __construct(PropertyMappingFactory $factory, MountManager $mountManager)
+    public function __construct(PropertyMappingFactory $factory, $registry)
```

Comment:
I think it would be better to keep the old parameter, making it optional, and add a third parameter. So, we can ensure BC.
Of course, this would require some logic to check if `MountManager` is passed directly or if it needs to be taken from registry

Reply:
AFAIK this is backward compatible in every ways and would allow us to deprecate the usage of MountManager if we want to in the future easily :) . I can change it to a third parameter of course, but I really think it's better suited here.

Reply:
You're right, it's already BC. But it forces to remove type hinting and doing manual checks on types.
I think that type hinting is more clear.
Feel free to add a deprecation here.

Reply:
My problem about the alternative (3rd argument) is that it can lead to situations in which you have two arguments populated at the same time. It would not make sense.

I understand your concern about the typehinting. However I think this is more a limitation of the language than an actual architecture issue (whereas having a third argument would have architecture consequences). 

Here is my suggestion: instead of an InvalidArgumentException, I can throw a TypeError. This way, when PHP will support union typehints, the migration towards them will be seemless and backward compatible. WDYT?

Reply:
It seems a reasonable compromise.

Reply:
And about the deprecation: OneUpFlysystemBundle should probably fix the issue first on their side, to avoid triggering deprecations without any ability to upgrade to a deprecation-free dependency IMO

Reply:
I did the change :)